tests/font-features.sil

```diff
@@ -0,0 +1,27 @@
+\begin[papersize=a5]{document}
+\script[src=packages/features]
```

Comment:
For unit tests, less is more. I've spend days and days reviewing them when things break and the fewer things being rolled into a test the better. Some tips:

* Use `\nofolios` and `\neverindent` to simplify both input and output.
* You aren't testing rules, don't load that package.
* You aren't testing rebox, don't use it.
* Don't include fancy Unicode like ✓ just because you can, use the absolute minimum characters / words to demonstrate the thing being tested.

Reply:
Would you perhaps like me to split it into multiple files? Since I'm testing multiple things.

First I'm testing if `Letters=SmallCaps` works, and if it's equivalent to the old way (`\add...`; `\remove...`) of using it.

Then I'm testing the needed multiple feature syntax I added, which I based off `fontspec.pdf`: `Ligatures=" { Historic, Discretionary }`, and making sure it's also equivalent to the old way.

Then I do a more complicated test, where I test nesting. I set up a parent `\font` with `CharacterVariant="{75,81}"`, and then in it put children `\font`s which do `CharacterVariant=No75`, `CharacterVariant=No81`. I then make sure that `\font[CharacterVariant="{No75, No81}"]` is exactly equivalent to that.

Finally, an unreleased version had a bug where `\font[size=3em]Test...` stopped working, because I had it wrapped in a `SILE.settings.temporarily`, which made the `\font` function stop working as expected when not fed content.

Reply:
Yes you can split it up, but the other points still apply. Also some things like that last point would be caught be other tests anyway (I hope) and so don't need a new test. The fewer unique tests that die when a specific feature breaks the easier it is to figure out why.

Reply:
Well no it won't be caught by another test unless that test loads `features.lua`, which none of them do. `features.lua` overwrites `\font` per recommendation of @simoncozens.

Reply:
And yes, what I'm saying is, if it's split up, suddenly we no longer need `rules.lua` to create visual separation. And we no longer need ✓ either because it's its own test so makes sense on its own without the prior semantics of "lines are equivalent".

Reply:
The test suite isn't primarily meant for visual inspection. Don't write tests with visual inspection in mind, write them primarily so the `-b debug` output can be diffed. Secondarily as a fallback when things go wrong there should be a visual clue about what failed.

Overwriting `\font` won't be an issue since we aren't going to do that, but still do test all the possible modes of your function (with/without content, and so forth). Just do so in as terse a way as possible so as to not trigger diffs for unrelated changes.